airflow_ext/gfw/operators/helper/flexible_operator.py

```diff
@@ -0,0 +1,51 @@
+from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+from airflow.operators.bash_operator import BashOperator
+
+import os
+import os.path
+
+
+class FlexibleOperator:
+    def __init__(self, parameters):
+        assert parameters
+        self.operator_parameters = parameters
+
+    def build_operator(self, kind):
+        if kind != 'bash' and kind != 'kubernetes':
+            raise ValueError('The operators allowed are bash or kubernetes, not <{}>'.format(kind))
+        assert self.operator_parameters['task_id']
+        assert self.operator_parameters['pool']
+        assert self.operator_parameters['cmds']
+        assert self.operator_parameters['image']
+        task_id = self.operator_parameters['task_id']
+        pool = self.operator_parameters['pool']
+        cmds = self.operator_parameters['cmds']
+        operator = None
+        if kind == 'bash':
+            assert self.operator_parameters['docker_run']
+            commands = '{} {} {}'.format(
+                self.operator_parameters['docker_run'],
+                self.operator_parameters['image'],
+                ' '.join(cmds))
+            # Left only the extra parameters that are not the bash_command
+            extra_params = { k : self.operator_parameters[k] for k in set(self.operator_parameters) - set(dict.fromkeys({'cmds','image','docker_run','name','dag'})) }
+            operator = BashOperator(
+                bash_command = commands,
+                **extra_params
+            )
+        else:
+            assert self.operator_parameters['name']
+            assert self.operator_parameters['dag']
+            # some pipelines has file run and not run.sh
+            run_path='./scripts/run.sh' if os.path.isfile('./scripts/run.sh') else './scripts/run'
```

Comment:
This is not needed. The problem is that we are using the `cmds` parameter on the `KubernetesPodOperator`, which boverrides the image `ENTRYPOINT` that is already pointing to the correct script on each pipeline. We should be using the `arguments` parameter instead.

I would set up an `arguments` parameter, and translate it to `commands`if it's a bash instance.

Reply:
@andres-arana thanks! that sounds good. I am making the change.